linfa-ensemble/src/random_forest/hyperparameters.rs

```diff
@@ -0,0 +1,80 @@
+use linfa_trees::DecisionTreeParams;
+
+#[derive(Clone, Copy)]
+pub struct RandomForestParams {
+    // number of decision trees to fit
+    pub n_estimators: usize,
+    // single tree hyperparameters
+    pub tree_hyperparameters: DecisionTreeParams,
+    // max number of features for the ensemble
+    pub max_features: u64,
+    // use all dataset or not
+    pub bootstrap: bool,
+}
+
+#[derive(Clone, Copy)]
+pub enum MaxFeatures {
+    Auto,
+    Sqrt,
+    Log2,
+}
+
+pub struct RandomForestParamsBuilder {
+    tree_hyperparameters: DecisionTreeParams,
+    n_estimators: usize,
+    max_features: Option<MaxFeatures>,
+    bootstrap: Option<bool>,
+}
+
+impl RandomForestParamsBuilder {
+    pub fn new(
+        tree_hyperparameters: DecisionTreeParams,
+        n_estimators: usize,
+    ) -> RandomForestParamsBuilder {
+        RandomForestParamsBuilder {
+            tree_hyperparameters,
+            n_estimators,
+            max_features: Some(MaxFeatures::Auto),
```

Comment:
If we initialise these to a (default) value then I think we don't need the `Option`. At the moment we sort of have two default values, one here and one in `build` when it does the `unwrap_or`.